src/coll/mod.rs

```diff
@@ -1271,11 +1277,14 @@ where
         }
 
         match cumulative_failure {
-            Some(failure) => Err(Error::new(
-                ErrorKind::BulkWrite(failure),
-                Some(error_labels),
-            )),
-            None => Ok(cumulative_result.unwrap_or_else(InsertManyResult::new)),
+            Some(mut failure) => {
+                failure.inserted_ids = cumulative_inserted_ids;
```

Comment:
This field is actually currently `pub(crate)`, and the only reason the user was able to observe it was the derived `Debug` implementation. I did some digging and found that RUST-260 was filed to eventually expose this field properly and to stop discarding the intermediate results, and I believe the field originally was introduced for counting writes attempted rather than actually tracking inserted ids (per https://github.com/mongodb/mongo-rust-driver/pull/377#discussion_r657342471). 

Given that we've done the work for populating this field, I think we probably ought to do RUST-260 also and mark the field as `pub`. 

Regarding the Jira logistics of this, I wonder if it makes sense to rename RUST-1488 to cover the bug of the `Debug` implementation leaking the unset, `pub(crate)` `inserted_ids` map, and instead have this PR be titled RUST-260. Wdyt?